alacritty_terminal/src/input.rs

```diff
@@ -198,10 +198,21 @@ impl<T: Eq> Binding<T> {
 
     #[inline]
     pub fn triggers_match(&self, binding: &Binding<T>) -> bool {
-        self.trigger == binding.trigger
-            && self.mods == binding.mods
-            && (self.mode.contains(binding.mode) || binding.mode.contains(self.mode))
-            && (self.notmode.contains(binding.notmode) || binding.notmode.contains(self.notmode))
+        // Check the binding's key and modifiers
+        if self.trigger != binding.trigger || self.mods != binding.mods {
+            return false;
+        }
```

Comment:
Just a small thing, but I think you could apply good ol' De Morgan's law to make this a bit more readable.

```
if self.trigger == binding.trigger && self.mods == binding.mods
```

This is also more similar to what we had written before.

Reply:
Can't early return if we do it that way though. That's the only reason why I've done it like this, otherwise you're 100% correct.

I thought it would be more readable to split this up into separate conditionals with early return statements than having just one giant boolean statement and trying to add comments to that.

Reply:
I mean, it still passes all the tests and is logically equivalent (see [De Morgan's Laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws).

I guess I forgot to add that you need to early return `true` and not `false`, but it's still possible.

Reply:
Generally this is even linted by clippy, but this is not accurate with an early return. So changing it would cause test failure.

The only alternative would be the following:
```rust
!(self.trigger == binding.trigger && self.mods == binding.mods)
```

And I don't think the introduction of brackets here is beneficial (it makes the whole thing a bit harder to read, since you need to read it inside-out).

Reply:
Yea, sorry for the noise... Today's a mess for me.